### ISSUE TITLE
Postinstall fails on pnpm fetch

### ISSUE THREAD

**Issue by Leksat** [netlify/cli#5928]
### Describe the bug

Latest version of `netlify-cli` fails on postinstall with `pnpm fetch`

```
$ pnpm fetch
...
.../node_modules/netlify-cli postinstall$ node ./scripts/postinstall.mjs
│ node:internal/modules/cjs/loader:1077
│   const err = new Error(message);
│               ^
│ Error: Cannot find module '@opentelemetry/core'
...
```

### Steps to reproduce

Repro: https://github.com/Leksat/netlify-cli-pnpm-fetch
Actions: https://github.com/Leksat/netlify-cli-pnpm-fetch/blob/main/.github/workflows/test.yml
Result: https://github.com/Leksat/netlify-cli-pnpm-fetch/actions/runs/5781917546/job/15667786482

### Environment

Local env:
```
  System:
    OS: macOS 13.5
    CPU: (12) arm64 Apple M2 Pro
    Memory: 128.36 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.17.0 - ~/.nvm/versions/node/v18.17.0/bin/node
    Yarn: 1.22.19 - ~/Library/pnpm/yarn
    npm: 9.6.7 - ~/.nvm/versions/node/v18.17.0/bin/npm
    pnpm: 8.6.12 - ~/.nvm/versions/node/v18.17.0/bin/pnpm
```
But the issue happens on Linux as well.

**Comment by Leksat** [comment]
According to my local tests, the issue was introduced in https://github.com/netlify/cli/releases/tag/v15.8.0

**Comment by dgp1130** [comment]
Ran into this myself today trying to use `netlify-cli` with pnpm and `hoist=false`. The root cause is that `@honeycombio/opentelemetry-node@0.4.0` had an undeclared dependency on `@opentelemetry/core`. This was fixed in https://github.com/honeycombio/honeycomb-opentelemetry-node/pull/207 and released in [`v0.5.0`](https://github.com/honeycombio/honeycomb-opentelemetry-node/releases/tag/v0.5.0). I'm pretty sure the fix here is just to bump the dependency in `@netlify/build`. I made https://github.com/netlify/build/pull/5247 with the fix.

In the mean time, the easiest workaround in pnpm is to use [`packageExtensions`](https://pnpm.io/package_json#pnpmpackageextensions) to add the undeclared dependency to your `package.json` file.

```json
{
    "pnpm": {
        "packageExtensions": {
            "@honeycomb/opentelemetry-node": {
                "dependencies": {
                    "@opentelemetry/core": "*"
                }
            }
        }
    }
}
```

**Comment by JGAntunes** [comment]
Hey folks 👋 Apologies for taking so long to tackle this one but I think this has been addressed with the latest CLI release - https://github.com/netlify/cli/blob/main/CHANGELOG.md#17131-2024-01-11

This packs - https://github.com/netlify/build/pull/5446 - which takes care of removing most of our OTEL dependencies from build execept for the OTEL API.

Feel free to re-open if this doesn't fix it for you.
